testing/src/scenario/context.py

```diff
@@ -365,6 +365,50 @@ def action(
             kwargs["id"] = id
         return _Event(f"{name}_action", action=_Action(name, **kwargs))
 
+    @staticmethod
+    def custom(
+        event: ops.BoundEvent,
+        *args: Any,
+        **kwargs: Any,
+    ):
+        """Event triggered by a charm library.
+
+        For example, for a library that emits a ``HappenedEvent`` event via a
+        ``MyConsumer`` object, and charm code like::
+
+            class MyCharm(ops.CharmBase):
+                def __init__(self, framework: ops.Framework):
+                    super().__init__(framework)
+                    self.source = MyConsumer(self, 'source')
+                    framework.observe(self.source.on.it_happened, self._on_happened_event)
+
+        Emit this event with::
+
+            ctx.run(ctx.on.custom(
+                MyConsumer.on.it_happened, 1, 2, foo='bar'
+            ), state)
+
+        Custom events do not have access to the Juju context of the originating
+        hook event.
+
+        Any additional arguments are passed through when instantiating the event
+        provided to observing handlers. Any of these arguments that are State
+        components (such as :attr:`ops.testing.Relation`) will be converted to
+        their ops counterparts (such as :attr:`ops.Relation`).
```

Comment:
Something seems off with the first sentence. How about this?

```suggestion
        Any additional arguments are passed through to the observing handlers
        when instantiating the event. Any of these arguments that are State
        components (such as :attr:`ops.testing.Relation`) will be converted to
        their ops counterparts (such as :attr:`ops.Relation`).
```

Reply:
"passed through to the observing handlers" sounds to me like you'd get a handler signature like `def _on_event(self, event, extra_arg1, extra_arg2)`. The extra arguments here are provided when creating the custom event (the event `__init__` looks like `def __init__(self, handle, extra_arg1, extra_arg2)`).

What about something like this?

```suggestion
        If the custom event takes more arguments than an :attr:`ops.Handle`,
        pass these as additional arguments to this method. Any of these arguments that are State
        components (such as :attr:`ops.testing.Relation`) will be converted to
        their ops counterparts (such as :attr:`ops.Relation`).
```

Reply:
Ah right, I get the point now. The handler would access the extra arguments as part of the event data. I like your suggestion. Would it be possible to change "this method" to `custom()`, to make the point even clearer?